core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerBase.java

```diff
@@ -755,13 +764,24 @@ public void cancel() {
       }
     }
 
-    private void trackNodeError(Node node, Throwable error) {
+    private void trackNodeError(Throwable error) {
       long latencyNanos = System.nanoTime() - this.start;
       context
           .getRequestTracker()
           .onNodeError(statement, error, latencyNanos, executionProfile, node);
     }
 
+    private void maybeUpdateSpeculativeExecutionsInFlightCounter(int amount) {
+      if (this.execution > 0) {
+        ((DefaultNode) node)
+            .getMetricUpdater()
+            .incrementCounter(
```

Comment:
Out of curiosity: what is the reason why we pass the profile name to all methods related to metrics? It's not used in the default implementation.

Reply:
I thought about splitting the metrics by execution profile at some point. I decided not to do it, but didn't want to completely close the door to the idea.